### PR TITLE
Show tax rates on order edit screen

### LINKED ISSUE #8242: 3.0 - Tax rate percentage / label not shown on admin orders page
### Expected behavior

When I view an order, I expect to see the tax percentage and the "label" (country / region from the tax rate)

### Actual behavior

The tax _amount_ is displayed, but not the percentage / label.

Screenshot from 2020-12-04 12-49-10

I'm filing this as a bug, because when you are in the middle of manually creating an order, the rate and label is shown:

Screenshot from 2020-12-04 12-52-12

But as soon as you save the order, that label is gone.

### Steps to reproduce the behavior
1) Make an order that has a tax rate applied. (Either manually or via the front-end; doesn't matter.)
2) View the order in the admin area.
3) Note the lack of tax rate.

### Information (if a specific version is affected):
PHP Version:

EDD Version (or branch):

WordPress Version:

Any other relevant information:

Related: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/8060

### PR DESCRIPTION
**Pull request by robincornett**
Fixes #8242

Proposed Changes:
1. When setting up the localized data for the order add/edit screen, only send an empty array of tax rate data to on the add edit screen, and only if taxes are enabled.
2. On the edit order screen, the tax rate ID, country, and region are localized.
3. Since the tax rate amount uses `floatval`, the template no longer restricts the rate to two decimal places (as this may not be accurate anyway).

This partially addresses #7799 as well--on the edit order screen, the tax rate is always shown if it exists, even if taxes are disabled on the store.

### CODE REVIEW

File: includes/admin/payments/orders.php [hunk 1]
```diff
@@ -684,6 +684,29 @@ function edd_order_details_overview( $order ) {
...
+			'region'  => '',
+		);
+	} elseif ( $tax_rate ) {
+		$has_tax = array(
+			'rate'    => $tax_rate,
+			'country' => '',
+			'region'  => '',
+		);
+		if ( $order->tax_rate_id ) {
+			$tax_rate_object = edd_get_adjustment( $order->tax_rate_id );
```
Comment (ashleyfae) [inline includes/admin/payments/orders.php:702]: https://github.com/easydigitaldownloads/easy-digital-downloads/pull/8253/files adds a new `get_tax_rate_object()` method so let's wait until that's merged, then this PR can be updated to use that.

Comment (robincornett) [reply includes/admin/payments/orders.php:702]: Oh, I can go ahead and update this PR--it's based off of that one. I did use the new `get_tax_rate` method in here already.

Comment (ashleyfae) [reply includes/admin/payments/orders.php:702]: Sure! :+1:

File: includes/admin/payments/orders.php [hunk 2]
```diff
@@ -684,6 +684,29 @@ function edd_order_details_overview( $order ) {
 		}
 	}
 
+	$has_tax  = 0;
```
Comment (spencerfinnell) [inline includes/admin/payments/orders.php:687]: Lets make the default `false` to match the first condition [here](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/admin/views/tmpl-order-totals.php#L32)

File: includes/admin/payments/orders.php [hunk 3]
```diff
@@ -684,6 +684,29 @@ function edd_order_details_overview( $order ) {
...
+	$tax_rate = $order->get_tax_rate();
+	if ( edd_is_add_order_page() && edd_use_taxes() ) {
+		$has_tax = array(
+			'rate'    => 0,
+			'country' => '',
+			'region'  => '',
+		);
```
Comment (spencerfinnell) [inline includes/admin/payments/orders.php:694]: Code style suggestion: maybe define the fallback outside of the condition so the same value can be used in multiple places.

**Review by spencerfinnell — CHANGES_REQUESTED** [review]

File: includes/admin/payments/orders.php (hunk 2 above)
Comment (robincornett) [reply includes/admin/payments/orders.php:687]: If I change it to `false`, it fails [this check](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/release/3.0/includes/admin/views/tmpl-order-totals.php#L29) instead (I think `false` is just empty after localization and `0` is read as `false`, at least as far as I can tell?).

Comment (spencerfinnell) [reply includes/admin/payments/orders.php:687]: Let's make it an explicit `false` string check so we don't have to figure out/guess what `wp_localize_script()` is doing.

Comment (robincornett) [reply includes/admin/payments/orders.php:687]: I'm okay with changing it but it does affect multiple templates (`order-totals`, `order-item`, `add-order-item` at least). Can I use `none`? I like that better than `no` for whatever reason.

Comment (spencerfinnell) [reply includes/admin/payments/orders.php:687]: Yep, that works.

**Comment by cklosowski** [comment]
I restarted the  failing tests here. They were just failing due to a timing issue by 1 second.

**Review by spencerfinnell — APPROVED** [review]

**Review by LisaCee — APPROVED** [review]
- [x] Country-wide tax rate
- [x] State tax rate
- [x] Refund
- [x] Tax rate and area visible after disabling taxes on Edit Order (#7799).

File: includes/admin/payments/class-refund-items-table.php [hunk 4]
```diff
@@ -432,7 +433,10 @@ public function display_rows() {
 			</td>
 		</tr>
 
-		<?php if ( edd_use_taxes() ) : ?>
+		<?php
+		$order = $this->get_order();
+		if ( $order && $order->get_tax_rate() ) :
+			?>
```
Comment (ashleyfae) [inline includes/admin/payments/class-refund-items-table.php:439]:
```suggestion
		?>
```

Minor indenting issue.

**Review by ashleyfae — CHANGES_REQUESTED** [review]
Tiny indenting issue - then I'll be happy to merge.

**Review by ashleyfae — APPROVED** [review]

**Pull request merged**